### PR TITLE
[INLONG-8967][Sort] Add Mysql connector on flink 1.15

### LINKED ISSUE #8967: [Feature] Add Mysql connector on flink 1.15
### Description

Parent issue: https://github.com/apache/inlong/issues/8336

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liaosunny123**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8967 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Comment by EMsnap** [comment]
Please add a end to end test on flink1.15 thx

**Comment by liaosunny123** [comment]
> Please add an end to end test on flink1.15 thx

ok and have tested in module flink end to end test.
image

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/utils/MySqlContainer.java [hunk 1]
```diff
@@ -60,8 +60,8 @@ protected Set<Integer> getLivenessCheckPorts() {
     @Override
     protected void configure() {
         // HERE is the difference, copy to /etc/mysql/, if copy to /etc/mysql/conf.d will be wrong
-        optionallyMapResourceParameterAsVolume(
-                MY_CNF_CONFIG_OVERRIDE_PARAM_NAME, "/etc/mysql/", "mysql-default-conf");
+        //optionallyMapResourceParameterAsVolume(
```
Comment (EMsnap) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/utils/MySqlContainer.java:63]: remove uselss code please

**Comment by EMsnap** [comment]
please add file settings in inlong-distribution/src/main/assemblies/sort-connectors-v1.15.xml

**Comment by liaosunny123** [comment]
> please add file settings in inlong-distribution/src/main/assemblies/sort-connectors-v1.15.xml

done

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/MysqlToRocksTest.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static String getNewStarRocksImageName() {
+        return NEW_STARROCKS_REPOSITORY + ":" + NEW_STARROCKS_TAG;
+    }
+
+    public static void buildStarRocksImage() {
```
Comment (EMsnap) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/MysqlToRocksTest.java:87]: duplicate with test code in postgreToStarrocks

Comment (liaosunny123) [reply inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/MysqlToRocksTest.java:87]: make a base class and reuse this logic ?

Comment (EMsnap) [reply inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/MysqlToRocksTest.java:87]: Yes it's one approach or you can create a class just for the initialization for starrocks and reuse it

**Review by EMsnap — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**